### ISSUE TITLE
DOI not found with 4.0.0 beta

### ISSUE THREAD

**Issue by bernhard-kleine** [JabRef/jabref#2780]
JabRef 4.0.0-beta
Windows 7 6.1 amd64 
Java 1.8.0_121
Steps to reproduce:

1. I have DOI 10.1016/j.neuron.2016.10.063
2. New Bibtex entry: I copy the id in the field
3. generate gives an error

```
Connection error
org.jabref.logic.importer.FetcherException: Connection error
	at org.jabref.logic.importer.fetcher.DoiFetcher.performSearchById(DoiFetcher.java:65)
	at org.jabref.gui.EntryTypeDialog$FetcherWorker.doInBackground(EntryTypeDialog.java:292)
	at org.jabref.gui.EntryTypeDialog$FetcherWorker.doInBackground(EntryTypeDialog.java:275)
	at javax.swing.SwingWorker$1.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at javax.swing.SwingWorker.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.io.IOException: Server returned HTTP response code: 504 for URL: http://data.crossref.org/10.1016%2Fj.neuron.2016.10.063
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.lang.reflect.Constructor.newInstance(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at org.jabref.logic.net.URLDownload.asString(URLDownload.java:152)
	at org.jabref.logic.net.URLDownload.asString(URLDownload.java:166)
	at org.jabref.logic.importer.fetcher.DoiFetcher.performSearchById(DoiFetcher.java:55)
	... 8 more
Caused by: java.io.IOException: Server returned HTTP response code: 504 for URL: http://data.crossref.org/10.1016%2Fj.neuron.2016.10.063
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at java.net.HttpURLConnection.getResponseCode(Unknown Source)
	at org.jabref.logic.net.URLDownload.openConnection(URLDownload.java:294)
	... 11 more
Put the excerpt of the log file here
```
I tested the same in 3.8.2 and it worked without changing my VPN connection.

**Comment by stefan-kolb** [comment]
This should be a temporary issue of the CrossRef service. Nothing that has to do with JabRef:
`Server returned HTTP response code: 504 for URL: http://data.crossref.org/10.1016%2Fj.neuron.2016.10.063`
We should implement a message for such occurences to signal an external unavailability to the user. See #2729

**Comment by bernhard-kleine** [comment]
I could get the DOI from JR 3.8.2 and I could not get it from JR4.0.0beta. I am fairly convinced the error is in jabref and not the server out of service. Did you check?

**Comment by stefan-kolb** [comment]
OK, thanks I'll investigate this further.

**Comment by Siedlerchr** [comment]
In the latest 4.0.0 dev snapshot from today 059f805e64da6ae3966aec018e81f908f54b5ed9 it works fine

**Comment by bernhard-kleine** [comment]
Are we supposed to go back to snapshots and do not keep the beta?

**Comment by koppor** [comment]
In case something does not work, please try the snapshot. Huge overhowl of
the entry editor is going in, but that should be done within the next few
weeks... Thank you for your patience!

Am 23.04.2017 16:57 schrieb "Bernhard Kleine" <notifications@github.com>:

**Comment by AEgit** [comment]
I can confirm that the DOI generation for 10.1016/j.neuron.2016.10.063 works with:
JabRef 4.0.0-dev--snapshot--2017-04-22--master--059f805e6
Windows 10 10.0 amd64 
Java 1.8.0_121

**Comment by matthiasgeiger** [comment]
I just tested it with the released beta version - and it worked fine.

However, we noticed that the crossref.org service that is used for the DOI-to-BibTeX feature is quite unstable in the last few weeks. So I suppose that it really was a temporily service outage on their side...

> Are we supposed to go back to snapshots and do not keep the beta?

Well, if we really found an issue and fixed it, then it would be useful to *go* ***on*** to the latest snapshots.
But as mentioned above: In this case there does not seem to be a general problem as none of the users/developers here could reproduce your problem.
